src/server/listener-manager.ts

```diff
@@ -0,0 +1,96 @@
+import { IDriveStorage } from '..';
+import { CacheEntry, Listener, ListenerStatus } from './types';
+
+export class ListenerManager {
+    private cache: CacheEntry[] = [];
+    private storage: IDriveStorage;
+
+    constructor(storage: IDriveStorage) {
+        this.storage = storage;
+    }
+
+    async addListener(listener: Listener) {
+        const syncUnits = await this.storage.getSyncUnits();
+        syncUnits.forEach(syncUnit => {
+            const { branch, documentId, scope, documentType } = syncUnit;
+            if (
+                listener.filter.branch.includes(branch) &&
```

Comment:
Let's move the filter check to a `checkFilter` method. We will also support setting the documentType as `*` or `makerdao/*`.